src/codeu/chat/client/commandline/Chat.java

```diff
@@ -14,310 +14,400 @@
+    return true;
+  }
 
-      if (!tokenScanner.hasNext()) {
-        System.out.println("ERROR: No user name supplied.");
-      } else {
-        signInUser(tokenScanner.nextLine().trim());
+  // CREATE ROOT PANEL
+  //
+  // Create a panel for the root of the application. Root in this context means
+  // the first panel and the only panel that should always be at the bottom of
+  // the panels stack.
+  //
+  // The root panel is for commands that require no specific contextual information.
+  // This is before a user has signed in. Most commands handled by the root panel
+  // will be user selection focused.
+  //
+  private Panel createRootPanel(final Context context) {
+
+    final Panel panel = new Panel();
+
+    // HELP
+    //
+    // Add a help command to the root panel. This will be called when the user
+    // enters "help" while on the root panel. This will print explanations for
+    // all commands that can be ran while on the root panel.
+    //
+    panel.register("help", new Panel.Command() {
+      @Override
+      public void invoke(Scanner args) {
+        System.out.println("ROOT MODE");
+        System.out.println("  u-list");
+        System.out.println("    List all users.");
+        System.out.println("  u-add <name>");
+        System.out.println("    Add a new user with the given name.");
+        System.out.println("  u-sign-in <name>");
+        System.out.println("    Sign in as the user with the given name.");
+        System.out.println("  exit");
+        System.out.println("    Exit the program.");
       }
-
-    } else if (token.equals("sign-out")) {
-
-      if (!clientContext.user.hasCurrent()) {
-        System.out.println("ERROR: Not signed in.");
-      } else {
-        signOutUser();
+    });
+
+    // U-LIST (user list)
+    //
+    // Add a command to print all users to the root panel. This will be called
+    // when the user entes "u-list" while on the root panel. This will print a
+    // list of all users registered on the server.
+    //
+    panel.register("u-list", new Panel.Command() {
+      @Override
+      public void invoke(Scanner args) {
+        for (final UserContext user : context.allUsers()) {
+          System.out.format(
+              "USER %s (%s)\n",
+              user.user.name,
+              user.user.id);
+        }
       }
-
-    } else if (token.equals("current")) {
-
-      showCurrent();
-
-    } else if (token.equals("u-add")) {
-
-      if (!tokenScanner.hasNext()) {
-        System.out.println("ERROR: Username not supplied.");
-      } else {
-        addUser(tokenScanner.nextLine().trim());
+    });
+
+    // U-ADD (add user)
+    //
+    // Add a command to add a new user to the root panel. This will be called when
+    // the user enters "u-add" while on the root panel. This will ask the server to
+    // create a new user with the given name. If successful, the user will be
+    // logged-in as the new user.
+    //
```

Comment:
This may be one of those polarizing points. From a user of the application stand-point I will always log in as the user I created. Think of any other time you created an account somewhere. That is why I chose that when you create a user, you would log-in as the user.